server/api/users/index.post.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * Handles the creation of a new user.
+ *
+ * This function validates the incoming user data against a predefined schema
+ * and creates a new user in the database if the validation is successful.
+ *
+ * @param {Event} event - The event object representing the HTTP request.
+ *
+ * @throws {Error} Throws an error if the user data is invalid.
+ *
+ * @returns {Object} An object containing a success message and the newly created user.
+ */
+import * as v from "valibot"
+import { createError, readValidatedBody } from "h3"
+import Users from "~~/server/database/models/User"
+
+export default defineEventHandler(async (event) => {
+  // Validate user data
+  const userSchema = v.object({
+    name: v.pipe(v.string(), v.nonEmpty("Please enter Name")),
+    email: v.pipe(v.string(), v.nonEmpty("Please enter Email"), v.email("Not an Email")),
+    password: v.pipe(v.string(), v.nonEmpty("Please enter Password"), v.minLength(8, "Password must be at least 8 characters"), v.regex(/[a-z]/, "Password must contain at least one lowercase letter"), v.regex(/[A-Z]/, "Password must contain at least one uppercase letter"), v.regex(/[0-9]/, "Password must contain at least one number")),
+  })
+
+  // This is a helper function that reads the body and validates it against the schema
+  const newUserContent = await readValidatedBody(event, body => v.parse(userSchema, body))
+  const validation = v.safeParse(userSchema, newUserContent)
+  if (!validation.success) {
+    throw createError({ status: 400, message: "Invalid user data" })
+  }
+
+  // Create new user
+  const newUser = await Users.create(newUserContent)
```

Comment:
Unfortunately, this does not work, since the role and verified fields also need to be set with default values. Also, somehow i get the error that the passwordHash is null, which it shouldn't be, but I don't know if the problem is in this pr or with the model.